mmeds/tools/tool.py

```diff
@@ -235,6 +237,100 @@ def summary(self):
+        """ Change anaconda3 environment """
+        # Some possible envs: 'qiime', 'pheniqs', 'mmeds'
+        full_env = env
+        if env == 'qiime':
+            full_env = 'qiime2-2020.8.0'
+        elif env == 'mmeds':
+            full_env = '/sc/arion/projects/MMEDS/admin_modules/mmeds-stable'
+        cmd = f'source activate {full_env};'
+        self.jobtext.append(cmd)
+
+    def conda_deactivate(self):
+        """ Deactivate current anaconda3 environment """
+        cmd = 'conda deactivate;'
+        self.jobtext.append(cmd)
+
+    def unzip_general(self, from_file, to_file):
+        """
+        A general unzip command for unzipping any artifact into a specified directory
+        """
+        # The -j and -o commands ignore the archive structure, meaning
+        #   all the files are extracted directly to the -d directory
+        cmd = 'unzip -jo {} -d {}'.format(from_file, to_file)
+        self.jobtext.append(cmd)
+
+    def move_general(self, from_file, to_file):
+        """
+        A general move for moving a file from one location to another
+        """
+        cmd = 'mv {} {}'.format(from_file, to_file)
+        self.jobtext.append(cmd)
+
+    def remove_general(self, rm_path):
+        """
+        A general remove for forcing the recursive deletion at a path
+        """
+        cmd = 'rm -rf {}'.format(rm_path)
+        self.jobtext.append(cmd)
+
+    def biom_convert(self, from_file, to_file, to_tsv=True, sanitize=True):
+        """
+        Perform a biom convert either to .biom or to .tsv
+        Requires biom installed in active environment
+        """
+        if to_tsv:
+            convert = '--to-tsv'
+        else:
+            convert = '--to-hdf5'
+
+        cmd = 'biom convert -i {} -o {} {}'.format(from_file,
+                                                   to_file,
+                                                   convert)
+        self.jobtext.append(cmd)
+
+        # Remove the biom convert quirk "Constructed from BIOM file" and leading '#'
+        if sanitize:
+            cmd = "sed -i '1d;2s/^#//' {}".format(to_file)
+            self.jobtext.append(cmd)
+
+    def extract_qiime2_feature_table(self, index_entry='filtered_table', clean=True):
+        """ Unzip qiime2 feature table artifact from previous analysis, extract its biom file, and convert to tsv """
+        self.add_path('tmp_feature_unzip')
+        self.add_path('biom_feature', '.biom')
+        self.add_path('feature_table', '.tsv')
+        # 'index_entry' must be a FeatureTable artifact
+        table = get_file_index_entry_location(self.path.parent, 'Qiime2', index_entry, self.testing)
+
+        self.unzip_general(table, self.get_file('tmp_feature_unzip'))
+        self.move_general(self.get_file('tmp_feature_unzip') / 'feature-table.biom', self.get_file('biom_feature'))
+        self.source_activate('qiime')
+        self.biom_convert(self.get_file('biom_feature'), self.get_file('feature_table'))
+        if clean:
```

Comment:
I know what clean means here but someone else might not. names should generally consist of a verb and a noun at least